.github/actions/build/action.yml

```diff
@@ -0,0 +1,40 @@
+name: 'build'
+description: 'Compiles and stores artifacts for further use'
+inputs:
+  relay-endpoint:
+    description: 'The endpoint of the relay e.g. relay.walletconnect.com'
+    required: false
+    default: 'relay.walletconnect.com'
+  project-id:
+    description: 'WalletConnect project id'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v3
+    
+    - uses: actions/cache@v3
+      with:
+        path: |
+          SourcePackagesCache
+        key: ${{ runner.os }}-spm-${{ hashFiles('**/Package.resolved') }}
```

Comment:
We using `actions/cache@v3` for caching DerivedData now (in same way with SourcePackagesCache). Why saving DerivedData separately in .tar archives is better?